Translation/Extractor/FileExtractor.php

```diff
@@ -53,8 +56,13 @@ public function __construct(\Twig_Environment $twig, LoggerInterface $logger, ar
         $this->twig = $twig;
         $this->logger = $logger;
         $this->visitors = $visitors;
-        $lexer = new \PHPParser_Lexer();
-        $this->phpParser = new \PHPParser_Parser($lexer);
+        $lexer = new Lexer();
+        if(class_exists('PhpParser\ParserFactory')) {
+            $factory = new ParserFactory();
+            $this->phpParser = $factory->create(ParserFactory::PREFER_PHP7,$lexer);
+        } else {
+            $this->phpParser = new \PHPParser_Parser($lexer);
+        }
 
```

Comment:
The seven lines above to create a parser is duplicated at least 4 times in this bundle. May I suggest to add a factory class with one static method that return a parser. 

'PhpParserFactory::create'